servers/rendering/rasterizer_rd/rasterizer_effects_rd.cpp

```diff
@@ -926,156 +926,337 @@ void RasterizerEffectsRD::bokeh_dof(RID p_base_texture, RID p_depth_texture, con
+	ssao.ssao_render_push_constant.inv_thickness_table[5] = inverse_range_factor / sample_thickness[5];
+	ssao.ssao_render_push_constant.inv_thickness_table[6] = inverse_range_factor / sample_thickness[6];
+	ssao.ssao_render_push_constant.inv_thickness_table[7] = inverse_range_factor / sample_thickness[7];
+	ssao.ssao_render_push_constant.inv_thickness_table[8] = inverse_range_factor / sample_thickness[8];
+	ssao.ssao_render_push_constant.inv_thickness_table[9] = inverse_range_factor / sample_thickness[9];
+	ssao.ssao_render_push_constant.inv_thickness_table[10] = inverse_range_factor / sample_thickness[10];
+	ssao.ssao_render_push_constant.inv_thickness_table[11] = inverse_range_factor / sample_thickness[11];
+
+	// These are the weights that are multiplied against the samples because not all samples are
+	// equally important.  The farther the sample is from the center location, the less they matter.
+	// We use the thickness of the sphere to determine the weight.  The scalars in front are the number
+	// of samples with this weight because we sum the samples together before multiplying by the weight,
+	// so as an aggregate all of those samples matter more.  After generating this table, the weights
+	// are normalized.
+	ssao.ssao_render_push_constant.sample_weight_table[0] = 4.0f * sample_thickness[0]; // Axial
+	ssao.ssao_render_push_constant.sample_weight_table[1] = 4.0f * sample_thickness[1]; // Axial
+	ssao.ssao_render_push_constant.sample_weight_table[2] = 4.0f * sample_thickness[2]; // Axial
+	ssao.ssao_render_push_constant.sample_weight_table[3] = 4.0f * sample_thickness[3]; // Axial
+	ssao.ssao_render_push_constant.sample_weight_table[4] = 4.0f * sample_thickness[4]; // Diagonal
+	ssao.ssao_render_push_constant.sample_weight_table[5] = 8.0f * sample_thickness[5]; // L-shaped
+	ssao.ssao_render_push_constant.sample_weight_table[6] = 8.0f * sample_thickness[6]; // L-shaped
+	ssao.ssao_render_push_constant.sample_weight_table[7] = 8.0f * sample_thickness[7]; // L-shaped
+	ssao.ssao_render_push_constant.sample_weight_table[8] = 4.0f * sample_thickness[8]; // Diagonal
+	ssao.ssao_render_push_constant.sample_weight_table[9] = 8.0f * sample_thickness[9]; // L-shaped
+	ssao.ssao_render_push_constant.sample_weight_table[10] = 8.0f * sample_thickness[10]; // L-shaped
+	ssao.ssao_render_push_constant.sample_weight_table[11] = 4.0f * sample_thickness[11]; // Diagonal
+
+	// If we aren't using all of the samples, delete their weights before we normalize.
+	if (!p_use_full_samples) {
+		ssao.ssao_render_push_constant.sample_weight_table[0] = 0.0f;
+		ssao.ssao_render_push_constant.sample_weight_table[2] = 0.0f;
+		ssao.ssao_render_push_constant.sample_weight_table[5] = 0.0f;
+		ssao.ssao_render_push_constant.sample_weight_table[7] = 0.0f;
+		ssao.ssao_render_push_constant.sample_weight_table[9] = 0.0f;
+	}
+
+	// Normalize the weights by dividing by the sum of all weights
+	float totalWeight = 0.0f;
```

Comment:
```suggestion
	float total_weight = 0.0f;
```
(Unless it's a direct copy from the MS code and you prefer to keep it similar.)